theano/gof/fg.py

```diff
@@ -478,8 +481,14 @@ def find_path_to(output_var, input_var):
 
         for node in new_nodes:
             assert node not in self.apply_nodes
-            self.__setup_node__(node)
-            self.apply_nodes.add(node)
+            prevent_addition = False
+            for n in self._removed_nodes :
+                if node is n :
+                    prevent_addition = True
+                    
+            if not prevent_addition : 
```

Comment:
@nouiz , if the errors are raised when an added node is trying to get added, wouldn't that cause some of the existing tests to fail(I was a little skeptical about this) ? I'm trying to add the nodes back only if they've not been added before. That's why ```if``` statement there. I'll give try what you've said and get back to you on what's happening